src/ui/reducers/layout.ts

```diff
@@ -27,35 +27,39 @@ const getDefaultSelectedPrimaryPanel = (session: any, recording?: Recording) =>
 };
 
 export async function getInitialLayoutState(): Promise<LayoutState> {
-  const recordingId = getRecordingId();
+  try {
+    const recordingId = getRecordingId();
 
-  // If we're in the library, there are no preferences to fetch.
-  if (!recordingId) {
-    return syncInitialLayoutState;
-  }
+    // If we're in the library, there are no preferences to fetch.
+    if (!recordingId) {
+      return syncInitialLayoutState;
+    }
+
+    const replaySessions = await asyncStore.replaySessions;
+    const recording = await getRecording(recordingId);
+    const session = replaySessions[recordingId!];
+
+    if (!session) {
+      return {
+        ...syncInitialLayoutState,
+        selectedPrimaryPanel: getDefaultSelectedPrimaryPanel(session, recording),
+      };
+    }
 
-  const replaySessions = await asyncStore.replaySessions;
-  const recording = await getRecording(recordingId);
-  const session = replaySessions[recordingId!];
+    const { viewMode, showVideoPanel, showEditor } = syncInitialLayoutState;
+    const initialViewMode = session.viewMode || viewMode;
+    trackEvent(initialViewMode == "dev" ? "layout.default_devtools" : "layout.default_viewer");
 
-  if (!session) {
     return {
       ...syncInitialLayoutState,
+      viewMode: initialViewMode,
       selectedPrimaryPanel: getDefaultSelectedPrimaryPanel(session, recording),
+      showVideoPanel: "showVideoPanel" in session ? session.showVideoPanel : showVideoPanel,
+      showEditor: "showEditor" in session ? session.showEditor : showEditor,
     };
+  } catch (e) {
+    return syncInitialLayoutState;
   }
```

Comment:
Why is this try/catch needed? It's scary that this could occasionally fail and we wouldnt know why

Reply:
You're right, I can make the `try/catch` much smaller.